src/main/kotlin/com/terraformation/backend/accelerator/db/VoteStore.kt

```diff
@@ -0,0 +1,101 @@
+import com.terraformation.backend.accelerator.model.toModel
+import com.terraformation.backend.auth.currentUser
+import com.terraformation.backend.db.accelerator.CohortPhase
+import com.terraformation.backend.db.accelerator.VoteOption
+import com.terraformation.backend.db.accelerator.tables.daos.ProjectVotesDao
+import com.terraformation.backend.db.accelerator.tables.pojos.ProjectVotesRow
+import com.terraformation.backend.db.accelerator.tables.references.PROJECT_VOTES
+import com.terraformation.backend.db.default_schema.ProjectId
+import com.terraformation.backend.db.default_schema.UserId
+import jakarta.inject.Named
+import java.time.InstantSource
+import org.jooq.Condition
+import org.jooq.DSLContext
+
+@Named
+class VoteStore(
+    private val clock: InstantSource,
+    private val dslContext: DSLContext,
+    private val projectVotesDao: ProjectVotesDao,
+) {
+  fun fetchAllVotesByProject(projectId: ProjectId): List<VoteModel> {
+    return fetch(PROJECT_VOTES.PROJECT_ID.eq(projectId))
+  }
+
+  fun create(model: VoteModel): VoteModel {
+    val now = clock.instant()
+    val currentUserId = currentUser().userId
+
+    val row =
+        ProjectVotesRow(
+            userId = model.userId,
+            projectId = model.projectId,
+            phaseId = model.phase,
+            voteOptionId = model.voteOption,
+            conditionalInfo = model.conditionalInfo,
+            createdBy = currentUserId,
+            createdTime = now,
+            modifiedBy = currentUserId,
+            modifiedTime = now)
+
+    projectVotesDao.insert(row)
+    return row.toModel()
+  }
+
+  fun delete(projectId: ProjectId, phase: CohortPhase? = null, userId: UserId? = null) {
+    dslContext.transaction { _ ->
+      val conditions = mutableListOf(PROJECT_VOTES.PROJECT_ID.eq(projectId))
+      if (phase != null) conditions.addLast(PROJECT_VOTES.PHASE_ID.eq(phase))
+      if (userId != null) conditions.addLast(PROJECT_VOTES.USER_ID.eq(userId))
+
+      val rowsDeleted = dslContext.deleteFrom(PROJECT_VOTES).where(conditions).execute()
+
+      if (rowsDeleted == 0) {
+        throw ProjectVoteNotFoundException(projectId, phase, userId)
+      }
+    }
+  }
+
+  fun updateVoteOption(
+      projectId: ProjectId,
+      phase: CohortPhase,
+      userId: UserId,
+      voteOption: VoteOption? = null,
+      conditionalInfo: String? = null
+  ) {
+    val rowsUpdated =
+        with(PROJECT_VOTES) {
+          dslContext
+              .update(PROJECT_VOTES)
+              .set(MODIFIED_BY, currentUser().userId)
+              .set(MODIFIED_TIME, clock.instant())
+              .set(VOTE_OPTION_ID, voteOption)
+              .set(CONDITIONAL_INFO, conditionalInfo)
+              .where(
+                  listOf(
+                      PROJECT_VOTES.PROJECT_ID.eq(projectId),
+                      PROJECT_VOTES.PHASE_ID.eq(phase),
+                      PROJECT_VOTES.USER_ID.eq(userId),
+                  ))
```

Comment:
This works, but just to keep it more syntactically similar to the raw SQL, we usually structure static lists of conditions like
```kotlin
.where(PROJECT_VOTES.PROJECT_ID.eq(project_id))
.and(PROJECT_VOTES.PHASE_ID.eq(phase))
.and(PROJECT_VOTES.USER_ID.eq(userId))
```